diff --git a/addons/point_of_sale/static/src/js/screens.js b/addons/point_of_sale/static/src/js/screens.js
index 517b433..37789db 100644
--- a/addons/point_of_sale/static/src/js/screens.js
+++ b/addons/point_of_sale/static/src/js/screens.js
@@ -1340,16 +1340,16 @@ function openerp_pos_screens(instance, module){ //module is instance.point_of_sa
                 });
 
             }else{
-                this.pos.push_order(currentOrder) 
-                if(this.pos.config.iface_print_via_proxy){
+                this.pos.push_order(currentOrder).then(function(){
+                if(self.pos.config.iface_print_via_proxy){
                     var receipt = currentOrder.export_for_printing();
-                    this.pos.proxy.print_receipt(QWeb.render('XmlReceipt',{
+                    self.pos.proxy.print_receipt(QWeb.render('XmlReceipt',{
                         receipt: receipt, widget: self,
                     }));
-                    this.pos.get('selectedOrder').destroy();    //finish order and go back to scan screen
+                    self.pos.get('selectedOrder').destroy();    //finish order and go back to scan screen
                 }else{
-                    this.pos_widget.screen_selector.set_current_screen(this.next_screen);
-                }
+                    self.pos_widget.screen_selector.set_current_screen(self.next_screen);
+                }})
             }
 
             // hide onscreen (iOS) keyboard 
